src/main/scala/deduplication/Deduplication.scala

```diff
@@ -181,15 +56,14 @@ class Deduplication[F[_]: Sync: Timer, ID, ProcessorID] private (
     * @param ifDuplicate The effect to run if the process is duplicate.
     * @return the result of [[ifNew]] or [[ifDuplicate]].
     */
-  def protect[A](id: ID, ifNew: F[A], ifDuplicate: F[A]): F[A] = {
+  final def protect[A](id: ID, ifNew: F[A], ifDuplicate: F[A])(implicit flatMap: FlatMap[F]): F[A] =
```

Comment:
I was able to keep the definition of `protect` in the trait, as its defined entirely in terms of other methods of `Deduplication`.